### DISCUSSION TITLE
SWC support

### DISCUSSION THREAD

**Discussion by wizpresso-steve-cy-fan** [sveltejs/kit#9664] [Ideas]
Despite Vite 4 "supports" SWC (and it is very vague at best) I did not see that SvelteKit is capable of supporting SWC for now? I searched online for guide but to no avail as well. Maybe we should consider adding support for SWC, the speedup is going to make a day-and-night difference in terms of development experience.

**Comment by tyzh-dev** [comment]
> Despite Vite 4 "supports" SWC (and it is very vague at best) I did not see that SvelteKit is capable of supporting SWC for now? I searched online for guide but to no avail as well. 

It could be supported through a Vite plugin. If it ever happens, a couple options:
- Offer a seperate @sveltejs/kit-swc/vite plugin instead and let users choose.
- Refactor the @sveltejs/kit/vite plugin and make SWC the default.
- Refactor the @sveltejs/kit/vite plugin and offer SWC as a setting flag.

> Maybe we should consider adding support for SWC, the speedup is going to make a day-and-night difference in terms of development experience.
- Luckily [Vite 4.3](https://vitejs.dev/blog/announcing-vite4-3.html) provides performance improvements. 
- Maybe SvelteKit could support TurboPack option as an extra addition in the future: https://github.com/sveltejs/kit/discussions/7432

Could you explain how refactoring the SvelteKit plugin to use SWC instead would meaningfully benefit the developer experience? Do you see any bottlenecks with the current implementation? Any concrete examples of it having a speedup compared to esbuild? Would it be worth the required time / maintenance cost to refactor the SvelteKit plugin?

**Reply by rohanrajpal** [reply]
one major pain we've seen is the time with "optimizing dependencies" specially if the library being used is large like "flowbite-svelte" or splidejs

not sure if swc will speed that up

**Reply by wizpresso-steve-cy-fan** [reply]
Good point. I already have used a plugin to replace ESBuild with SWC, so while the baseline performance does not have marginal difference, let me head it up with a project that have more dependencies.

**Reply by rohanrajpal** [reply]
Oh sure, can you help me out on how we can test it ourselves too?

**Reply by rohanrajpal** [reply]
@MAYANK-M31 can you also help with more list of dependcies that get optmized with vite dev? That might help checking this out faster

**Reply by dmdin** [reply]
I have already tested [SWC](https://www.npmjs.com/package/unplugin-swc) with Vite to enable reflect-metadata, because esbuild [doesn't support it](https://github.com/evanw/esbuild/issues/257). And it worked perfectly for me. I assume, that different bundlers must be supported to prevent things happened like with esbuild if the author stops support future TypeScript versions. 

Also it would be perfect if the build-time was much faster. It's starting to be a problem for huge projects to run test and build scripts, that can take several minutes to finish. In this case [Turbopack](https://vercel.com/templates/svelte/turborepo-sveltekit-starter) and [Rolldown](https://github.com/rolldown-rs/) seem to be promising.

**Reply by MrBns** [reply]
if svelte-5 could do that.. it would be a magical event...  React already supporting it for a long time.. we want two beast (swc+svelte) together... please do it.

**Reply by rohanrajpal** [reply]
Just to be clear, we are looking to have something like https://github.com/vitejs/vite-plugin-react-swc available, right?
